### ISSUE TITLE
facet_grid does not allow single variable facets

### ISSUE THREAD

**Issue by wcbeard** [yhat/ggpy#183]
I'm wondering why `facet_grid` raises the exception when `x` or `y` isn't passed. I see the same can be accomplished with `facet_wrap`, but I think R's `facet_grid` allows a single variable (like 1st example in [the docs](http://docs.ggplot2.org/0.9.3.1/facet_grid.html)).

```
import ggplot as g

df = g.diamonds
df = df[df.color.isin(['I', 'J'])]

# This breaks:
p = g.ggplot(g.aes(x='carat', y='price'), data=df) + g.geom_point()
p + g.facet_grid(x='color')
~/src/ggplot/ggplot/geoms/facet_grid.py in __init__(self, x, y, scales)
      7         self.y = y
      8         if self.x is None or self.y is None:
----> 9             raise Exception("facet_grid(): need both x and y mapping! Use facet_wrap() for only one dimension.")
     10         self.ncol = None
     11         self.nrow = None

Exception: facet_grid(): need both x and y mapping! Use facet_wrap() for only one dimension.


# This works
p2 = p + g.facet_wrap(y='color')
g.ggsave('plots/facet.pdf', plot=p2)
```

The p2 plot results as follows:

![facet](https://f.cloud.github.com/assets/379413/2013447/5c01a546-8795-11e3-8448-c5766d794204.png)

Changing `facet_wrap(y='color')` to `facet_wrap(x='color')` doesn't give 2 columns, but changing `nrows` (actually should be `ncols`, #182) does:

```
p3 = p + g.facet_wrap(x='color', nrow=2)
g.ggsave('plots/facet2.png', plot=p3)
```

![facet2](https://f.cloud.github.com/assets/379413/2013505/846f4078-8796-11e3-81ad-00774828717b.png)

When I try to manually trick `facet_grid`, it also breaks, so it looks like it would involve more than just eliminating the Exception:

```
facet = g.facet_grid(x='color', y='dummy')
facet.y = None
p + facet

~/src/ggplot/ggplot/geoms/facet_grid.py in __radd__(self, gg)
     22             y = gg.data.get(self.y)
     23         except:
---> 24             raise Exception("facet_grid(): mapping for y (\"%s\") is not available in the DataFrame" % self.y)
     25         if x is None:
     26             n_dim_x = 1

Exception: facet_grid(): mapping for y ("None") is not available in the DataFrame

```

**Bottom line**: `facet_wrap` allows for single-variable subplots with control over # rows and columns, but I believe it would make sense for `facet_grid` to allow this kind of operation as well.
